subsys/net/l2/ieee802154/ieee802154.c

```diff
@@ -119,6 +119,54 @@ static inline void set_pkt_ll_addr(struct net_linkaddr *addr, bool comp,
 	addr->type = NET_LINK_IEEE802154;
 }
 
+/**
+ * Filters the destination address of the frame (used when IEEE802154_HW_FILTER
+ * is not available).
+ */
+static bool ieeee802154_check_dst_addr(struct net_if *iface,
+				      struct ieee802154_mhr *mhr)
+{
+	struct ieee802154_context *ctx = net_if_l2_data(iface);
+	const struct ieee802154_radio_api *radio =
+		net_if_get_device(iface)->api;
+
+	if (radio && (radio->get_capabilities(net_if_get_device(iface)) &
+		      IEEE802154_HW_FILTER)) {
+		return true;
+	}
+
+	if (mhr->fs->fc.dst_addr_mode == IEEE802154_ADDR_MODE_NONE) {
+		return true;
+
+	} else if (mhr->fs->fc.dst_addr_mode == IEEE802154_ADDR_MODE_EXTENDED) {
+		if (mhr->dst_addr->plain.pan_id == 0xFFFF) {
+			return true; // it's the broadcast
+		}
+		uint8_t is_broadcast = 1; // until first non-0xFF encountered
+		uint8_t i;
+		uint8_t *frame_addrp = mhr->dst_addr->plain.addr.ext_addr;
+		uint8_t *expected_addrp = ctx->ext_addr;
+		for (i=0; i<IEEE802154_MAX_ADDR_LENGTH; i++) {
```

Comment:
Both symbols have the same value, but I'd rather see `IEEE802154_EXT_ADDR_LENGTH` here.